src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int SCALE_FACTOR = 2;
+    private Node<K, V>[] table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+    private int capacity = DEFAULT_CAPACITY;
+    private int threshold = (int) (DEFAULT_LOAD_FACTOR * DEFAULT_CAPACITY);
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        int tableIndex = getIndexForKey(key);
+        if (table[tableIndex] == null) {
+            table[tableIndex] = new Node<>(key, value);
+        } else {
+            Node<K, V> node = table[tableIndex];
+            Node<K, V> lastNode;
+            do {
+                if (Objects.equals(node.key, key)) {
+                    node.value = value;
+                    return;
+                }
+                lastNode = node;
+                node = node.next;
+            } while (node != null);
+            lastNode.next = new Node<>(key, value);
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int tableIndex = getIndexForKey(key);
+        if (table[tableIndex] == null) {
+            return null;
+        }
+        Node<K, V> node = table[tableIndex];
+        do {
```

Comment:
do you really need `do while` , not `while` loops? Don't use them without a reason